pkg/apiserver/resthandler.go

```diff
@@ -234,7 +256,7 @@ func ListResource(r rest.Lister, rw rest.Watcher, scope RequestScope, forceWatch
 		opts := *out.(*api.ListOptions)
```

Comment:
Done.

Also, I would like to avoid spending too much time on it, because it will be significantly simplified once we migrate to unversioned.ListOptions:
https://github.com/kubernetes/kubernetes/pull/17369/files#diff-fe6dbe5fa548ea5c031a5be1669a6376R242